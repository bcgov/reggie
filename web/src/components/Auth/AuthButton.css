.auth-button {
  position: relative;
  margin-left: auto;
  padding: 8px 16px;
  border: none;
  background-color: #fcba19;
  color: #003366;
  text-transform: uppercase;
  letter-spacing: 0.2em;
  border-radius: 2px;
  outline-width: 0;
  text-decoration: none;
  overflow: hidden;
  cursor: pointer;
  -webkit-transition-duration: 0.4s; /* Safari */
  transition-duration: 0.4s;
}

@media screen and (max-width: 500px) {
  .auth-button {
    letter-spacing: .1em;
    padding: 4px 8px;
    font-size: 0.8rem;
  }
}
