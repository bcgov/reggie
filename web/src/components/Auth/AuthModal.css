.auth-modal.modal {
  display: flex !important;
  align-items: center;
  background-color: white;
}

.modal-header {
  background-color: #003366;
  border-bottom: 2px solid #fcba19;
  color: #fff;
  height: 100px;
  padding: 0.5rem 1rem !important;
}

.modal-title {
  width: 400px;
}

.auth-modal .modal-footer {
  display: block;
}

.auth-modal .auth-buttons {
  display: flex;
  justify-content: space-evenly;
}

.auth-modal .auth-button {
  margin: 0;
}

.modal-body {
  height: 100px;
}

@media screen and (max-width: 500px) {
  .auth-modal .modal-dialog {
    width: 100%;
    margin: 0;
  }
}
