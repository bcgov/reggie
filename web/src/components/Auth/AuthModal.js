//
// Reggie Web
//
// Copyright Â© 2018 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Jason Leach on 2018-10-03.
//

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import Banner from '../UI/Banner';
import { SSO_IDP, APP_INFO } from '../../constants';
import './AuthModal.css';

export const AuthModal = ({ isAuthenticated }) => {
  return (
    <Modal modalClassName="auth-modal" isOpen={!isAuthenticated} fade={false}>
      <ModalHeader>
        <Banner titleText={APP_INFO.NAME} />
      </ModalHeader>
      <ModalBody>Please login before starting Rocket.Chat Invitation with Reggie</ModalBody>
      <ModalFooter>
        <p>Login with:</p>
        <div className="auth-buttons">
          <Link className="auth-button" to={{ pathname: '/login', state: { idp: SSO_IDP.GITHUB } }}>
            GitHub
          </Link>
          <Link className="auth-button" to={{ pathname: '/login', state: { idp: SSO_IDP.IDIR } }}>
            IDIR
          </Link>
        </div>
      </ModalFooter>
    </Modal>
  );
};

AuthModal.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
};

export default AuthModal;
