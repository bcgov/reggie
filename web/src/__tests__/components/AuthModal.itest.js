import React from 'react';
import { shallow } from 'enzyme';
import { AuthModal } from '../../components/Auth/AuthModal';

describe('AuthButton Component', () => {
  it('matches snapshot when Logged in', () => {
    const authState = true;
    const wrapper = shallow(<AuthModal isAuthenticated={authState} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when Logged out', () => {
    const authState = false;
    const wrapper = shallow(<AuthModal isAuthenticated={authState} />);
    expect(wrapper).toMatchSnapshot();
  });
});
