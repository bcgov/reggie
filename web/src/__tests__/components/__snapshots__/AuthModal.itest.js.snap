// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton Component matches snapshot when Logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthModal
    isAuthenticated={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoFocus": true,
      "backdrop": true,
      "backdropTransition": Object {
        "mountOnEnter": true,
        "timeout": 150,
      },
      "centered": false,
      "children": Array [
        <ModalHeader
          charCode={215}
          closeAriaLabel="Close"
          tag="h5"
          wrapTag="div"
        >
          <Banner
            titleText="Pathfinder Rocket.Chat Invitation App"
          />
        </ModalHeader>,
        <ModalBody
          tag="div"
        >
          Please login before starting Rocket.Chat Invitation with Reggie
        </ModalBody>,
        <ModalFooter
          tag="div"
        >
          <p>
            Login with:
          </p>
          <div
            className="auth-buttons"
          >
            <Link
              className="auth-button"
              replace={false}
              to={
                Object {
                  "pathname": "/login",
                  "state": Object {
                    "idp": "github",
                  },
                }
              }
            >
              GitHub
            </Link>
            <Link
              className="auth-button"
              replace={false}
              to={
                Object {
                  "pathname": "/login",
                  "state": Object {
                    "idp": "idir",
                  },
                }
              }
            >
              IDIR
            </Link>
          </div>
        </ModalFooter>,
      ],
      "fade": false,
      "isOpen": false,
      "keyboard": true,
      "modalClassName": "auth-modal",
      "modalTransition": Object {
        "timeout": 300,
      },
      "onClosed": [Function],
      "onOpened": [Function],
      "role": "dialog",
      "zIndex": 1050,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "charCode": 215,
          "children": <Banner
            titleText="Pathfinder Rocket.Chat Invitation App"
          />,
          "closeAriaLabel": "Close",
          "tag": "h5",
          "wrapTag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "titleText": "Pathfinder Rocket.Chat Invitation App",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Please login before starting Rocket.Chat Invitation with Reggie",
          "tag": "div",
        },
        "ref": null,
        "rendered": "Please login before starting Rocket.Chat Invitation with Reggie",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <p>
              Login with:
            </p>,
            <div
              className="auth-buttons"
            >
              <Link
                className="auth-button"
                replace={false}
                to={
                  Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "github",
                    },
                  }
                }
              >
                GitHub
              </Link>
              <Link
                className="auth-button"
                replace={false}
                to={
                  Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "idir",
                    },
                  }
                }
              >
                IDIR
              </Link>
            </div>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login with:",
            },
            "ref": null,
            "rendered": "Login with:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="auth-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "github",
                      },
                    }
                  }
                >
                  GitHub
                </Link>,
                <Link
                  className="auth-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "idir",
                      },
                    }
                  }
                >
                  IDIR
                </Link>,
              ],
              "className": "auth-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "GitHub",
                  "className": "auth-button",
                  "replace": false,
                  "to": Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "github",
                    },
                  },
                },
                "ref": null,
                "rendered": "GitHub",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "IDIR",
                  "className": "auth-button",
                  "replace": false,
                  "to": Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "idir",
                    },
                  },
                },
                "ref": null,
                "rendered": "IDIR",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <ModalHeader
            charCode={215}
            closeAriaLabel="Close"
            tag="h5"
            wrapTag="div"
          >
            <Banner
              titleText="Pathfinder Rocket.Chat Invitation App"
            />
          </ModalHeader>,
          <ModalBody
            tag="div"
          >
            Please login before starting Rocket.Chat Invitation with Reggie
          </ModalBody>,
          <ModalFooter
            tag="div"
          >
            <p>
              Login with:
            </p>
            <div
              className="auth-buttons"
            >
              <Link
                className="auth-button"
                replace={false}
                to={
                  Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "github",
                    },
                  }
                }
              >
                GitHub
              </Link>
              <Link
                className="auth-button"
                replace={false}
                to={
                  Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "idir",
                    },
                  }
                }
              >
                IDIR
              </Link>
            </div>
          </ModalFooter>,
        ],
        "fade": false,
        "isOpen": false,
        "keyboard": true,
        "modalClassName": "auth-modal",
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "role": "dialog",
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "charCode": 215,
            "children": <Banner
              titleText="Pathfinder Rocket.Chat Invitation App"
            />,
            "closeAriaLabel": "Close",
            "tag": "h5",
            "wrapTag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "titleText": "Pathfinder Rocket.Chat Invitation App",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Please login before starting Rocket.Chat Invitation with Reggie",
            "tag": "div",
          },
          "ref": null,
          "rendered": "Please login before starting Rocket.Chat Invitation with Reggie",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Login with:
              </p>,
              <div
                className="auth-buttons"
              >
                <Link
                  className="auth-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "github",
                      },
                    }
                  }
                >
                  GitHub
                </Link>
                <Link
                  className="auth-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "idir",
                      },
                    }
                  }
                >
                  IDIR
                </Link>
              </div>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login with:",
              },
              "ref": null,
              "rendered": "Login with:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    className="auth-button"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/login",
                        "state": Object {
                          "idp": "github",
                        },
                      }
                    }
                  >
                    GitHub
                  </Link>,
                  <Link
                    className="auth-button"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/login",
                        "state": Object {
                          "idp": "idir",
                        },
                      }
                    }
                  >
                    IDIR
                  </Link>,
                ],
                "className": "auth-buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "GitHub",
                    "className": "auth-button",
                    "replace": false,
                    "to": Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "github",
                      },
                    },
                  },
                  "ref": null,
                  "rendered": "GitHub",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "IDIR",
                    "className": "auth-button",
                    "replace": false,
                    "to": Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "idir",
                      },
                    },
                  },
                  "ref": null,
                  "rendered": "IDIR",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`AuthButton Component matches snapshot when Logged out 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthModal
    isAuthenticated={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoFocus": true,
      "backdrop": true,
      "backdropTransition": Object {
        "mountOnEnter": true,
        "timeout": 150,
      },
      "centered": false,
      "children": Array [
        <ModalHeader
          charCode={215}
          closeAriaLabel="Close"
          tag="h5"
          wrapTag="div"
        >
          <Banner
            titleText="Pathfinder Rocket.Chat Invitation App"
          />
        </ModalHeader>,
        <ModalBody
          tag="div"
        >
          Please login before starting Rocket.Chat Invitation with Reggie
        </ModalBody>,
        <ModalFooter
          tag="div"
        >
          <p>
            Login with:
          </p>
          <div
            className="auth-buttons"
          >
            <Link
              className="auth-button"
              replace={false}
              to={
                Object {
                  "pathname": "/login",
                  "state": Object {
                    "idp": "github",
                  },
                }
              }
            >
              GitHub
            </Link>
            <Link
              className="auth-button"
              replace={false}
              to={
                Object {
                  "pathname": "/login",
                  "state": Object {
                    "idp": "idir",
                  },
                }
              }
            >
              IDIR
            </Link>
          </div>
        </ModalFooter>,
      ],
      "fade": false,
      "isOpen": true,
      "keyboard": true,
      "modalClassName": "auth-modal",
      "modalTransition": Object {
        "timeout": 300,
      },
      "onClosed": [Function],
      "onOpened": [Function],
      "role": "dialog",
      "zIndex": 1050,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "charCode": 215,
          "children": <Banner
            titleText="Pathfinder Rocket.Chat Invitation App"
          />,
          "closeAriaLabel": "Close",
          "tag": "h5",
          "wrapTag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "titleText": "Pathfinder Rocket.Chat Invitation App",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Please login before starting Rocket.Chat Invitation with Reggie",
          "tag": "div",
        },
        "ref": null,
        "rendered": "Please login before starting Rocket.Chat Invitation with Reggie",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <p>
              Login with:
            </p>,
            <div
              className="auth-buttons"
            >
              <Link
                className="auth-button"
                replace={false}
                to={
                  Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "github",
                    },
                  }
                }
              >
                GitHub
              </Link>
              <Link
                className="auth-button"
                replace={false}
                to={
                  Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "idir",
                    },
                  }
                }
              >
                IDIR
              </Link>
            </div>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login with:",
            },
            "ref": null,
            "rendered": "Login with:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="auth-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "github",
                      },
                    }
                  }
                >
                  GitHub
                </Link>,
                <Link
                  className="auth-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "idir",
                      },
                    }
                  }
                >
                  IDIR
                </Link>,
              ],
              "className": "auth-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "GitHub",
                  "className": "auth-button",
                  "replace": false,
                  "to": Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "github",
                    },
                  },
                },
                "ref": null,
                "rendered": "GitHub",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "IDIR",
                  "className": "auth-button",
                  "replace": false,
                  "to": Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "idir",
                    },
                  },
                },
                "ref": null,
                "rendered": "IDIR",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <ModalHeader
            charCode={215}
            closeAriaLabel="Close"
            tag="h5"
            wrapTag="div"
          >
            <Banner
              titleText="Pathfinder Rocket.Chat Invitation App"
            />
          </ModalHeader>,
          <ModalBody
            tag="div"
          >
            Please login before starting Rocket.Chat Invitation with Reggie
          </ModalBody>,
          <ModalFooter
            tag="div"
          >
            <p>
              Login with:
            </p>
            <div
              className="auth-buttons"
            >
              <Link
                className="auth-button"
                replace={false}
                to={
                  Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "github",
                    },
                  }
                }
              >
                GitHub
              </Link>
              <Link
                className="auth-button"
                replace={false}
                to={
                  Object {
                    "pathname": "/login",
                    "state": Object {
                      "idp": "idir",
                    },
                  }
                }
              >
                IDIR
              </Link>
            </div>
          </ModalFooter>,
        ],
        "fade": false,
        "isOpen": true,
        "keyboard": true,
        "modalClassName": "auth-modal",
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "role": "dialog",
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "charCode": 215,
            "children": <Banner
              titleText="Pathfinder Rocket.Chat Invitation App"
            />,
            "closeAriaLabel": "Close",
            "tag": "h5",
            "wrapTag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "titleText": "Pathfinder Rocket.Chat Invitation App",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Please login before starting Rocket.Chat Invitation with Reggie",
            "tag": "div",
          },
          "ref": null,
          "rendered": "Please login before starting Rocket.Chat Invitation with Reggie",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Login with:
              </p>,
              <div
                className="auth-buttons"
              >
                <Link
                  className="auth-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "github",
                      },
                    }
                  }
                >
                  GitHub
                </Link>
                <Link
                  className="auth-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "idir",
                      },
                    }
                  }
                >
                  IDIR
                </Link>
              </div>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login with:",
              },
              "ref": null,
              "rendered": "Login with:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    className="auth-button"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/login",
                        "state": Object {
                          "idp": "github",
                        },
                      }
                    }
                  >
                    GitHub
                  </Link>,
                  <Link
                    className="auth-button"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/login",
                        "state": Object {
                          "idp": "idir",
                        },
                      }
                    }
                  >
                    IDIR
                  </Link>,
                ],
                "className": "auth-buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "GitHub",
                    "className": "auth-button",
                    "replace": false,
                    "to": Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "github",
                      },
                    },
                  },
                  "ref": null,
                  "rendered": "GitHub",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "IDIR",
                    "className": "auth-button",
                    "replace": false,
                    "to": Object {
                      "pathname": "/login",
                      "state": Object {
                        "idp": "idir",
                      },
                    },
                  },
                  "ref": null,
                  "rendered": "IDIR",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
